org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/NonScalaSourceErrorMarkersTest.scala

```diff
@@ -0,0 +1,81 @@
+package org.scalaide.core
+package sbtbuilder
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IProjectDescription
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.junit.Assert
+import org.junit.Ignore
+import org.junit.Test
+import org.scalaide.core.SdtConstants
+import org.scalaide.ui.internal.preferences.ScalaPluginSettings
+import org.scalaide.util.internal.SettingConverterUtil
+
+import testsetup.Implicits
+import testsetup.SDTTestUtils
+
+class NonScalaSourceErrorMarkersTest {
+  import SDTTestUtils._
+  import Implicits.TestableProject
+
+  def addScalaNature(project: IProject) = natures(project) { description =>
+    SdtConstants.NatureId +: description.getNatureIds
+  }
+
+  private def natures(project: IProject)(newNatures: IProjectDescription => Array[String]): Unit = {
+    val description = project.getDescription
+    description.setNatureIds(newNatures(description))
+    project.setDescription(description, null)
+  }
+
+  def removeScalaNature(project: IProject) = natures(project) { description =>
+    description.getNatureIds.filter { _ != SdtConstants.NatureId }
+  }
+
+  def soThen(assertThat: => Unit): Unit = assertThat
```

Comment:
I don't see the point of this method, on the other hand it doesn't really matter if it is there or not.